Objective-C/CBLDatabase.mm

```diff
@@ -624,9 +655,30 @@ - (nullable NSDate*) getDocumentExpirationWithID: (NSString*)documentID {
 #pragma mark - INTERNAL
 
 - (void) mustBeOpen {
-    if (_c4db == nullptr) {
+    if ([self isClosed])
         [NSException raise: NSInternalInconsistencyException
                     format: @"%@", kCBLErrorMessageDBClosed];
+}
+
+- (void) mustBeOpenLocked {
+    CBL_LOCK(self) {
+        [self mustBeOpen];
+    }
+}
+
+- (void) mustBeOpenAndNotClosing {
+    if ([self isClosed] || _isClosing)
+        [NSException raise: NSInternalInconsistencyException
+                    format: @"%@", kCBLErrorMessageDBClosed];
+}
+
+- (BOOL) isClosed {
+    return _c4db == nullptr;
+}
+
+- (BOOL) isClosedLocked {
```

Comment:
same with `isClosed`. seems all accessors are using the lock, which means we can make base method `isClosed` thread safe, and remove locked method.

Reply:
I want to separate it out because we don't want to re-lock again.